src/plan_computation.py

```diff
@@ -58,4 +58,78 @@ def generate_photo_plan_on_grid(
         Scan plan as a list of waypoints.
 
     """
-    raise NotImplementedError()
\ No newline at end of file
+    max_distance = compute_distance_between_images(camera, dataset_spec)
+    min_speed = compute_speed_during_photo_capture(camera, dataset_spec)
+    waypoints = []
+    x_points = math.ceil(dataset_spec.scan_dimension_x / max_distance[0])
+    y_points = math.ceil(dataset_spec.scan_dimension_y / max_distance[1])
+    curr_x = 0
+    curr_y = 0
+    for i in range(0, y_points):
+        if i % 2 == 0:
+            for j in range(0, x_points):
+                waypoints.append(Waypoint(curr_x, curr_y, dataset_spec.height, min_speed))
+                curr_x += max_distance[0]
+            curr_y += max_distance[1]
+            curr_x -= max_distance[0]
+        else:
+            for j in range(x_points):
+                waypoints.append(Waypoint(curr_x, curr_y, dataset_spec.height, min_speed))
+                curr_x -= max_distance[0]
+            curr_y += max_distance[1]
+            curr_x += max_distance[0]
+    return waypoints
+
+
+def each_flight_time_computation(distance: float, velocity: float, acceleration: float, start_speed: float, end_speed: float) -> float:
+    """Compute the time required to fly a certain distance with given velocity and acceleration constraints.
+
+    Args:
+        distance: The distance to be flown (in meters).
+        velocity: The maximum velocity of the drone (in m/s).
+        acceleration: The acceleration and deceleration rate of the drone (in m/s^2).
+        start_speed: The speed at the start of the segment (in m/s).
+        end_speed: The speed at the end of the segment (in m/s).
+
+    Returns:
+        The time required to fly the given distance (in seconds).
+    """
+    if distance == 0:
+        return 0.0
+    time_to_accelerate = (velocity - start_speed) / acceleration
+    time_to_decelerate = (velocity - end_speed) / acceleration
+    distance_to_accelerate = 0.5 * velocity * time_to_accelerate
```

Comment:
This is not correct. You have to include the full area under the speed v time plot. There is a triangle which vertical edge (v_max - start_speed) and horizontal edge time_to_accelerate, and below it is a rectangle with dimensions time_to_accelerate and start_speed.